### DISCUSSION TITLE
Insert variables in the document metadata (top YAML section)

### DISCUSSION THREAD

**Discussion by e-alizadeh** [quarto-dev/quarto-cli#2865] [Q&A] [answered]
I've created a blog using Quarto. If possible, I'd like to use variables in the document metadata. In particular, in order to standardize the name of categories, I want to use variables defined somewhere else like in `_variables.yml` file and then call it in a post metadata. 
Going through the [Variables](https://quarto.org/docs/authoring/variables.html) page, it seems that it's not possible since Quarto treats that section differently. So, I thought of asking here if there is a way to use variables in the document metadata? 

If not, then I can create a feature request!

**Comment by mcanouil** [comment]
I am not sure to understand what you try/want to achieve, could you provide an example, even if those are not working as expected?

**Reply by e-alizadeh** [reply]
Sorry. Here is an example. 

Let's say I have the following `_variables.yml` file next to `_quarto.yml`:

```yaml
kws:
  - k1: keyword1
  - k2: keyword2
  - k3: keyword3
```

And I want to read those variables in the markdown's meta section as below:
```md
---
title: "title"
categories: [{{ <var kws.k1 >}}, {{ <var kws.k2 >}}]
abc: def
---

# Header 1

Reading {{ <var kws.k1 >}} works!
```

**Comment by mcanouil** [comment, marked as answer]
The issue is that you are using an array to define the keywords and to call the var shortcode.

<details><summary>See the code</summary>

- Quarto document:

  ````
  ---
  title: "title"
  categories:
    - "{{< var kws.k1 >}}"
    - "{{< var kws.k2 >}}"
  abc: def
  ---
  
  # Header 1
  
  Reading {{< var kws.k1 >}} works!
  ````

- `_variables.yml`
  ```
  kws:
    k1: keyword1
    k2: keyword2
    k3: keyword3
  ```

</details>

**Reply by e-alizadeh** [reply]
Thanks @mcanouil . Your code works. Thanks for the help.

**Reply by nithinmkp** [reply]
can you explain this a little bit? i did not understand how quarto is understanding  kws.k1 is coming from _variables.yml

**Reply by mcanouil** [reply]
That's the var shortcode, see the documentation.

I suggest you navigate through the documentation instead of reviving two years old thread.

**Comment by wiesehahn** [comment]
Is there also a way to insert paths?



E.g. in `_variables.yml` I define 
```
favicon: /images/favicon.png
```
and add it in `_quarto.yml` with
```
website:
    favicon: "{{< var favicon >}}"
```
it reads the path as `path_to_project/_site/{{< var favicon >}}'`

**Reply by mcanouil** [reply]
@wiesehahn I opened an issue to allow markdown parsing:

- #6455

**Comment by Beforerr** [comment]
Is it possible to insert dict-like variables? For example I want to insert an author across mulitple documents.

- `_variables.yml`

```
authors:
  abc:
     name: ABC 
     email: abc@abc.com
     orcid: 0000-0000
```

- in document
```
---
title: "title"
authors:
  - "{{< var authors.abc >}}"
---

**Reply by mcanouil** [reply]
Did you try it?
Note that you still need to provide the correct schema for authors.

**Reply by Beforerr** [reply]
I tried it and it did not work

**Reply by mcanouil** [reply]
Then you have your answer.

Why not use `_quarto.yml`, `_metadata.yml`, and/or YAML frontmatter for this?
And possibly Quarto profiles?

And or `metadata-files` key to provide an `_authors.yml` or whatever.

**Reply by Beforerr** [reply]
The reason for not using `_quarto.yml`, `_metadata.yml` is that I need it in some documents but not all documents. 
Also nothing appear if I use variables in the main body of a document.

```
Reading "{{< var authors.abc >}}" does not work:)
```

**Reply by mcanouil** [reply]
Then use a global default one and override on a per document basis.
